src/ralph/deployment/deployment.py

```diff
@@ -438,15 +459,93 @@ def check_ip_from_defined_network(address):
 
 
 @deployment_action(
+    verbose_name=_('Assign new IP'),
+    form_fields={
+        'network': {
+            'field': forms.ChoiceField(
+                label=_('IP Address')
+            ),
+            'choices': next_free_ip_choices,
+            'exclude_from_history': True
+        },
+    },
+    run_after=['assign_new_hostname'],
+)
+def assign_new_ip(cls, instances, network, **kwargs):
+    for instance in instances:
+        network = Network.objects.get(pk=network)
+        ip = network.issue_next_free_ip()
+        logger.info('Assigning {} to {}'.format(ip, instance))
+        ethernet = Ethernet.objects.create(base_object=instance)
+        logger.info('Bounding {} to {} ethernet'.format(ip, ethernet))
+        ip.ethernet = ethernet
+
+        try:
+            hostname = kwargs['shared_params']['hostnames'].get(instance.pk)
+            if hostname:
+                ip.hostname = hostname
+        except KeyError:
+            pass
+
+        ip.save()
+
+
+def base_object_ip_choices(actions, objects):
+    ipaddresses = []
+    for instance in objects:
+        for ip in instance.ipaddresses:
+            ipaddresses.append(
+                [ip.id, '{} - {}'.format(ip.address, ip.hostname)]
+            )
+    return ipaddresses
+
+
+def base_object_network_choices(actions, objects):
+    networks = []
+    for instance in objects:
+        for network in instance._get_available_networks():
+            networks.append(
+                [network.id, network]
+            )
+    return networks
+
+
+@deployment_action(
+    verbose_name=_('Replace IP'),
+    form_fields={
+        'ipaddress': {
+            'field': forms.ChoiceField(
+                label=_('IP Address'),
+            ),
+            'choices': base_object_ip_choices,
+            'exclude_from_history': True,
+        },
+        'network': {
+            'field': forms.ChoiceField(
+                label=_('Network'),
+            ),
+            'choices': base_object_network_choices,
+            'exclude_from_history': True,
+        },
+    }
+)
+def replace_ip(cls, instances, ipaddress, network, **kwargs):
+    ip = IPAddress.objects.get(pk=ipaddress)
+    network = Network.objects.get(pk=network)
+    ip.address = str(network.get_first_free_ip())
```

Comment:
some logging will be useful here (ex. 'Replacing <OLD-IP> to <NEW-IP> for <HOSTNAME>')